web/netlify/functions/update-settings.ts

```diff
@@ -0,0 +1,45 @@
+import { Handler } from "@netlify/functions";
+import { verifyTypedData } from "viem";
+import { createClient } from "@supabase/supabase-js";
+import messages from "../../src/consts/eip712-messages";
+
+const SUPABASE_KEY = process.env.SUPABASE_CLIENT_API_KEY;
+const SUPABASE_URL = process.env.SUPABASE_URL;
+const supabase = createClient(SUPABASE_URL!, SUPABASE_KEY!);
+
+export const handler: Handler = async (event) => {
+  try {
+    if (!event.body) {
+      throw new Error("No body provided");
+    }
+    // TODO: sanitize event.body
+    const { email, telegram, nonce, address, signature } = JSON.parse(event.body);
+    const lowerCaseAddress = address.toLowerCase() as `0x${string}`;
+    // Note: this does NOT work for smart contract wallets, but viem's publicClient.verifyMessage() fails to verify atm.
```

Comment:
yea that's unfortunate. publicClient.verifyMessage() worked when this was done? hoping that's a temporary error because we should also support smart contract wallets

Reply:
No solution for this right now.